EX5-1: Create a program that opens the English text file "week5-test.txt", print each line of string, and close the file.
• Make sure to remove the line feed code at the end of each line before printing.
• Also remember to always execute close() function after processing.

In [ ]:
#Exercise 5-1
f = open('week5-test.txt')
for line in f:
  striped = line.strip() #this is removing the line feed code
  print(striped)
f.close()

abcd
1234


EX5-2: Write a program that opens the english text file "week5-test.txt" again and print
each line of string along with its corresponding line numbers in the following format:
• 1 abcd
• 2 1234

In [ ]:
#Exercise 5-2
i=0
f = open('week5-test.txt')
for line in f:
  i+=1 #trying to get the number of the line
  striped = line.strip() #this is removing the line feed code
  print(i, striped)
f.close()

1 abcd
2 1234


EX5-3: Write a program that opens the English text file "icuhistory.txt" and prints the first 10 lines, each with its corresponding line number.

In [24]:
#Exercise 5-3
i=0
f = open('icuhistory.txt')
for line in f:
  i+=1
  striped = line.strip() #this is removing the line feed code
  if i==10+1:
    break
  print(i, striped)

f.close()

1 in the fall of 1945 just weeks after the end
2 of world war ii a group of japanese christian educators
3 initiated a move to establish a university based on christian
4 principles the foreign missions conference of north america and the
5 us education mission both visiting japan at the time
6 gave their wholehearted support conveying this plan to people in
7 the us amidst the post-war yearning for reconciliation and
8 world peace americans supported this project with great enthusiasm in
9 1948 the japan international christian university foundation jicuf was
10 established in new york to coordinate fund-raising efforts in the


EX5-4: Create a program that reopens “icuhistory.txt" and prints only the lines that contain the string "icu".

Hint: Use the function on slide p.12 to determine if it the string is present. You can determine whether a line contains the string by checking if the return value is -1 or not.
• Even if “icu” is part of a larger string (e.g. “jicuf”), consider it a match and print the line.

In [27]:
#Exercise 5-4
f = open('icuhistory.txt')
for line in f:
  striped = line.strip()
  matchplace = line.find("icu") #determining which line has 'icu'
  if matchplace!=-1: #limiting the printing lines to what contains 'icu'
    print(striped)
f.close()

1948 the japan international christian university foundation jicuf was
lead the national fund raising drive icu has been unique from
in 1950 icu purchased a spacious site in mitaka city
in march 1953 the japanese ministry of education authorized icu
science and technology selected icu s research and education
icu for its fiscal year 2007 distinctive university education support
icu is continually working to reconfirm its responsibilities and fulfill


EX5-5: Write a program that reopens "icuhistory.txt" and prints the total number of words and the number of unique words in the text.
• You can create a word list using the method on p.14 and calculate the total word count and the number of unique words.

In [30]:
#Exercise 5-5
wordlist=[]
count=0 #the number for counting all the words
count_u=0 #the number for counting all the unique words

f = open('icuhistory.txt')
for line in f:
  striped = line.strip()
  words = striped.split(" ")
  for word in words:
    wordlist.append(word) #generating a wordlist with all words from the file

for element in wordlist: #counting number of words in the wordlist
  count+=1

uniquewordlist=set(wordlist) #eliminating the duplicates
for element in uniquewordlist:
  count_u+=1 #counting the number of elements in the uniquewordlist
print('The number of words:', count)
print('The number of unique words:', count_u)
f.close()

The number of words: 548
The number of unique words: 274


EX5-6: Create a dictionary where each word in “icuhistory.txt” is a key, and the value is the number of times the word appears.
• You can either use the word list from EX5-5 to create a dictionary or reopen “icuhistory.txt” and recreate the dictionary.

In [44]:
#Exercise 5-6
diction={}
wordlist=[]
count=0
f = open('icuhistory.txt')
for line in f:
  striped = line.strip()
  words = striped.split(" ")
  for word in words:
    wordlist.append(word) #generating a wordlist with all words from the file

for element in wordlist: #counting number of words in the wordlist
  count+=1

for i in set(wordlist): #for all the elements in the list that eliminated the duplicates
  diction[i]=wordlist.count(i) #the value of key is the number of its appearance
print(diction)
f.close()

{'': 1, 'laid': 1, 'selected': 1, 'foundation': 2, 'project': 1, 'through': 2, 'missions': 1, '29': 1, 'make': 1, 'both': 1, 'america': 1, '2007': 1, '31': 1, '2004': 1, 'the': 40, 'tokyo': 1, 'within': 1, 'learning': 1, 'acknowledgement': 1, '1953': 1, 'four-year': 1, 'organization': 2, 'mission': 2, 'inauguration': 1, 'as': 4, 'excellence': 1, 'ambitious': 1, 'christian': 6, 'major': 1, 'good': 2, 'advising': 1, 'at': 2, 'divisions': 2, 'visiting': 1, 'dream': 1, 'projects': 1, 'unique': 1, 'conducted': 1, 'was': 3, 'procured': 1, 'war': 1, 's': 2, 'has': 1, 'international': 3, 'a': 13, 'during': 1, 'set': 1, 'gave': 1, 'students': 3, 'fund': 1, 'events': 1, 'plan': 2, 'school': 2, 'icu': 6, 'great': 1, 'culture': 1, 'by': 1, 'transcending': 1, 'start': 1, 'were': 1, 'leading': 1, 'decision': 1, 'half': 1, 'york': 1, 'year': 4, 'enforced': 1, 'also': 2, 'supporting': 1, 'on': 5, 'center': 1, 'down': 1, 'authorized': 1, 'while': 1, 'majors': 1, 'after': 1, 'their': 5, 'an': 3, 'in': 1

Answer EX5-7 to EX5-9 below for "scorelist.txt", which contains the student ID numbers and scores for 40 students.
Each line contains "121787 74" with the ID and score separated by spaces.
No answer needed: Open "scorelist.txt" and print the student ID number and the score, respectively.

EX5-7: Create and print a dictionary (e.g. scoredict) in which the student number is the key and the score is the value.
• Convert the scores to numbers (int) when reading them in.

In [65]:
#Exercise 5-7
scoredict={}

f = open('scorelist.txt')
for line in f:
  striped = line.strip()
  ID = striped.split(' ')
  scoredict[ID[0]]=int(ID[1]) #the first value of each list is the key of the dictionary; its value is equal to the second element of the line
print(scoredict)
f.close()

{'121787': 74, '121367': 71, '121817': 88, '121619': 85, '131445': 80, '131244': 96, '131872': 98, '131963': 75, '131172': 78, '131965': 72, '131112': 90, '131956': 87, '141105': 61, '141703': 61, '141407': 78, '141569': 82, '141585': 89, '141455': 82, '141370': 80, '141837': 67, '141857': 86, '141497': 94, '141853': 67, '141245': 80, '151452': 83, '151238': 62, '151827': 58, '151409': 40, '151789': 95, '151742': 71, '151133': 40, '151095': 49, '151186': 75, '151586': 51, '151926': 73, '151975': 96, '151079': 49, '151091': 100, '151588': 49, '151630': 61}


EX5-8: Calculate the average score of 40 people. Either of the following is acceptable:
• Calculate the total score and the number of people while creating a dictionary with EX5-7 and finally calculate the average
• After creating EX5-7 dictionary, use the list of values in the dictionary

In [69]:
#Exercise 5-8
scoredict={}
sum=0
f = open('scorelist.txt')
for line in f:
  striped = line.strip()
  ID = striped.split(' ')
  scoredict[ID[0]]=int(ID[1]) #the first value of each list is the key of the dictionary; its value is equal to the second element of the line
  sum+=int(ID[1]) #the total of all the values, which is the score
print('The average score is:', sum/40) #calculating the average
f.close()

The average score is: 74.325


EX5-9: Create a dictionary whose student ID numbers are the keys and their grades are the values, and print it.
Additionally, print the number of people who received each grade ('A', 'B', 'C', 'D', 'E')

In [94]:
#Exercise 5-9
scoredict={}

f = open('scorelist.txt')
for line in f:
  striped = line.strip()
  ID = striped.split(' ')
  scoredict[ID[0]]=int(ID[1]) #the first value of each list is the key of the dictionary; its value is equal to the second element of the line

  if int(ID[1])>=90:
    scoredict[ID[0]]="A"
  elif int(ID[1])>=80:
    scoredict[ID[0]]="B"
  elif int(ID[1])>=70:
    scoredict[ID[0]]="C"
  elif int(ID[1])>=60:
    scoredict[ID[0]]="D"
  else:
    scoredict[ID[0]]="F"
print(scoredict)
f.close()

scorelist= []
for value in scoredict.values(): #turn the values of the dictionary into a list
  scorelist.append(value)
  scorelist=sorted(scorelist) #sort it alphabetically

freq={}
for grade in scorelist: #generating a dictionary counting frequency
  freq[grade]=scorelist.count(grade)
print(freq)

{'121787': 'C', '121367': 'C', '121817': 'B', '121619': 'B', '131445': 'B', '131244': 'A', '131872': 'A', '131963': 'C', '131172': 'C', '131965': 'C', '131112': 'A', '131956': 'B', '141105': 'D', '141703': 'D', '141407': 'C', '141569': 'B', '141585': 'B', '141455': 'B', '141370': 'B', '141837': 'D', '141857': 'B', '141497': 'A', '141853': 'D', '141245': 'B', '151452': 'B', '151238': 'D', '151827': 'F', '151409': 'F', '151789': 'A', '151742': 'C', '151133': 'F', '151095': 'F', '151186': 'C', '151586': 'F', '151926': 'C', '151975': 'A', '151079': 'F', '151091': 'A', '151588': 'F', '151630': 'D'}
{'A': 7, 'B': 11, 'C': 9, 'D': 6, 'F': 7}


EX5-10: Create a Diary Management App You are tasked with creating a diary management app to record daily events. The user will be able to add diary entries and search or view saved entries in a simple application.

Requirements:
1. The user can add new diary entries.
2. Each diary entry is saved in a file with both the date and content.
3. The user can view all saved diary entries.
4. The user can search for diary entries by a specific date and view the content.

Features to Implement:
1. Add New Diary Entry
The user should be able to add a new diary entry. The app should automatically include the current date along with the content
entered by the user. The entry should then be saved in a file.
● Save format: YYYY-MM-DD : Diary content
● Use append mode to add entries to the file.
2. View All Diary Entries
The user should be able to view all previously saved diary entries. Read all lines from the file and display them in a list format.
● If the file does not exist or no entries have been added yet, display the message: "No diary entries found."
3. Search for Diary Entries by Date
The user should be able to search for saved diary entries by specifying a date (in YYYY-MM-DD format). If an entry is found, display
its content.
● If no entry is found for the specified date, display the message: "No entries found for that date."
● If the file does not exist, show the same message.
4. Exit the App
Implement a feature that allows the user to exit the app. When the user exits, display the message: "Exiting the diary management
tool." 

In [107]:
#Exercise 5-10

from datetime import datetime # importing the current date
current_datetime = datetime.now().strftime("%Y-%m-%d")

content = input('Add a new diary entry: ') #inputing the entry of the diary

wf = open('diary.txt', 'w')
wf.write(f'Date: {current_datetime}\n') #printing the date of the day
wf.write(f'Diary entry: {content}\n') #printing the entry after a line
wf.close()


print(current_datetime)
if len(content) ==0: #checks whether the file contains entries
  print('No diary entries found.')

else:
  f=open('diary.txt','r')
  for line in f:
    striped = line.strip()
    print(striped)
  print('Diary entry added.')

while True:
  print('\n1. Add new entry: ')
  print('2. View all entries')
  print("3. Search for Diary Entries by Date")
  print("4. Exit")
  choice = input('Choose an option(1-4): ')

  if choice=='1':
    current_datetime = datetime.now().strftime("%Y-%m-%d")
    content = input('Add a new diary entry: ') #inputing the entry of the diary
    wf = open('diary.txt', 'a') #appends the entry to the file
    wf.write(f'Date: {current_datetime}\n') #printing the date of the day
    wf.write(f'Diary entry: {content}\n') #printing the entry after a line

    print(current_datetime)
    if len(content) ==0: #checks whether the file contains entries
      print('No diary entries found.')

    else:
      wf=open('diary.txt','r') #reads the file
      for line in f:
        striped = line.strip()
        print(striped)
      print('Diary entry added.')

  elif choice=='2':
    if len(content)==0: #checks whether the file contains entries
      print('No diary entries found.')

    else:
      f=open('diary.txt','r') #reads the file
      for line in f:
        striped = line.strip()
        print(striped) #prints each striped line from the file

  elif choice=='3':
    if len(content)>0:
      wf=open('diary.txt', 'r')
      search = input('Enter the date to search(YYYY-MM-DD): ')
      found=False

      for entry in wf:
        if current_datetime==search: #checking from each line in the file if the input matches anything
          print(entry.strip())  #outputing that entry
          found=True
      if found==False:
        print('No entries found for that date.')

    else:
      print("No entries found for that date.")

  elif choice=='4':
    print("Exiting the diary management tool.")
    break #exiting the while loop
  else:
    print('Please inpute number 1-4') #for the entries that have no destination within the loop

Add a new diary entry: 
2024-10-12
No diary entries found.

1. Add new entry: 
2. View all entries
3. Search for Diary Entries by Date
4. Exit
Choose an option(1-4): 2
No diary entries found.

1. Add new entry: 
2. View all entries
3. Search for Diary Entries by Date
4. Exit
Choose an option(1-4): 1
Add a new diary entry: hello
2024-10-12
lo
Diary entry added.

1. Add new entry: 
2. View all entries
3. Search for Diary Entries by Date
4. Exit
Choose an option(1-4): 3
Enter the date to search(YYYY-MM-DD): 2024-10-12
Date: 2024-10-12
Diary entry:
Date: 2024-10-12
Diary entry: hello

1. Add new entry: 
2. View all entries
3. Search for Diary Entries by Date
4. Exit
Choose an option(1-4): 4
Exiting the diary management tool.
